### PR TITLE
Fix deprecated gcloud compute networks --mode switches.

### LINKED ISSUE #54238: cluster/gce uses deprecated gcloud parameter "mode"
/kind bug

**What happened**:

Warning printed by `e2e-up.sh`: `WARNING: mode is deprecated. Please use subnet-mode instead.`

**What you expected to happen**:

No warning.

**How to reproduce it (as minimally and precisely as possible)**:

`kubetest -v --up`

**Anything else we need to know?**:

This is my first e2e cluster. Logs with more context below.

**Environment**:
- Kubernetes version (use `kubectl version`): master
- Cloud provider or hardware configuration**: GCE
- OS (e.g. from /etc/os-release): Description:	Ubuntu 14.04 LTS

```
2017/10/19 16:18:51 util.go:154: Running: ./hack/e2e-internal/e2e-up.sh
Project: k8s-scale-testing
Network Project: k8s-scale-testing
Zone: us-east1-x
... Starting cluster in us-east1-x using provider gce
... calling verify-prereqs
... calling verify-kube-binaries
... calling kube-up
Project: k8s-scale-testing
Network Project: k8s-scale-testing
Zone: us-east1-x
+++ Staging server tars to Google Storage: gs://kubernetes-staging-6524d3db95/e2e-test-porridge-devel
+++ kubernetes-server-linux-amd64.tar.gz uploaded (sha1 = 2b91b5046dc30dd52e7cdea6086191b16a02bb56)
+++ kubernetes-salt.tar.gz uploaded (sha1 = 3e7907f15509540958c02a427aff989a340b8601)
+++ kubernetes-manifests.tar.gz uploaded (sha1 = ef5169fc331f2d2fafaf37a7995bd0367f1eb11e)
INSTANCE_GROUPS=
NODE_NAMES=
Looking for already existing resources
Creating new auto network: e2e-test-porridge
WARNING: mode is deprecated. Please use subnet-mode instead.
Created [https://www.googleapis.com/compute/v1/projects/k8s-scale-testing/global/networks/e2e-test-porridge].
NAME               SUBNET_MODE  BGP_ROUTING_MODE  IPV4_RANGE  GATEWAY_IPV4
e2e-test-porridge  AUTO         REGIONAL
```

### PR DESCRIPTION
**Pull request by rmmh**
"create --mode" becomes "create --subnet-mode", and switch-mode has been
folded into "update".

Create --mode was deprecated in October and will be removed in the next
gcloud release. It is already failing in staging tests.
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** Fixes #54238

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -1675,7 +1675,7 @@ function create-network() {
 
 function expand-default-subnetwork() {
   gcloud compute networks switch-mode "${NETWORK}" \
-    --mode custom \
```
Comment (ixdy) [inline cluster/gce/util.sh]: this one is still `--subnet-mode`.

Comment (ixdy) [reply cluster/gce/util.sh]: but also
```
SYNOPSIS
    gcloud compute networks switch-mode NAME --mode=MODE [GCLOUD_WIDE_FLAG ...]

DESCRIPTION
    (DEPRECATED) switch-mode is deprecated. Please use
    --switch-to-custom-subnet-mode with gcloud compute networks update instead.

    gcloud compute networks switch-mode is used to change the mode of a
    network. Currently, only changing from auto to custom mode is supported.
```

Comment (rmmh) [reply cluster/gce/util.sh]: Done!

**Comment by krzyzacy** [comment]
gke test need a kubetest bump, sending out

**Comment by krzyzacy** [comment]
/retest
gce looks like flaky

**Comment by ixdy** [comment]
hmm:
```
failed to create Stackdriver Logging sink: googleapi: Error 400: There is a maximum of 90 sinks allowed, failedPrecondition
```

**Comment by cblecker** [comment]
/assign @gmarek
for approval

**Comment by liggitt** [comment]
are these changes needed in release branches as well?

**Comment by krzyzacy** [comment]
@liggitt yep (this will not be blocking for gce until this gcloud staging version come to prod, and we bump gcloud in e2e image)

**Comment by krzyzacy** [comment]
/assign @mikedanese @roberthbailey 
can someone approve?

**Comment by roberthbailey** [comment]
Do we need to backport this to any old release branches to keep the upgrade tests working?

**Comment by tpepper** [comment]
@roberthbailey see #54238 for the over arching issue status.  Indication there is yes on backports.

**Comment by krzyzacy** [comment]
yep, we don't bind kubernetes with specific gcloud versions, each active branch should be able to run with latest gcloud.

**Pull request merged**